### PR TITLE
Restructure application structure 

### PR DESCRIPTION
**Pull request by ZachEddy**
# Restructure Application Structure 

This pull request restructures how applications are bootstrapped by the Atlas CLI. It also includes a few minor housekeeping changes, like using the publicly-available version of the protoc gentool.

Summary of the major changes:
- Move `svc/` and `pb/` into a new `pkg/` folder.
- Create a `db/` folder, which is empty for now. In the near future, the `/db` folder will be conditionally added by setting a `-db` flag to `true`.
- Add a test file for the `zserver.go` file.
- Add a `make clean` target that removes all app-related Docker images.
- Use the public-facing protoc gentool.

### CODE REVIEW

**Comment by johnbelamaric** [comment]
`config` doesn't belong in `cmd`. Maybe it will go away anyway once we build the configuration package.

**Comment by dmacthedestroyer** [comment]
@johnbelamaric  that's the plan

File: cli/atlas/templates/pkg/svc/zserver_test.go.gotmpl [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+	var tests = []struct {
+		expected *pb.VersionResponse
+		err      error
+	}{
+		{
+			expected: &pb.VersionResponse{Version: version},
+			err:      nil,
+		},
+	}
+	for _, test := range tests {
```
Comment (dmacthedestroyer) [inline cli/atlas/templates/pkg/svc/zserver_test.go.gotmpl]: I love the idea of providing a default test file to show the proper convention for building tests in Go. It turns out we've been doing them wrong, however: https://golang.org/pkg/testing/#hdr-Subtests_and_Sub_benchmarks

wrap the body of this `for` loop in a `t.Run` call so we do it the way the docs do it.

Comment (ZachEddy) [reply cli/atlas/templates/pkg/svc/zserver_test.go.gotmpl]: Good find!

File: cli/atlas/templates/pkg/svc/zserver.go.gotmpl [hunk 2]
```diff
@@ -24,20 +24,25 @@ import (
 // TODO: Update the GetVersion function when newer versions of your service
 // become available. Feel free to change GetVersion to better-suit how your
 // versioning system, or get rid of it entirely. GetVersion helps make up a
-// simple "starter" example that allows an end-to-end example. It is not
+// simple "starter" example that allows an end-to-end example. It is not`
```
[the remark this answers is not in the document]
Comment (ZachEddy) [reply cli/atlas/templates/pkg/svc/zserver.go.gotmpl]: Ugh. Fixed it!

**Pull request merged**